src/App.js

```diff
@@ -9,102 +9,152 @@ import Footer from "./Footer.js";
 import Alert from "./Alert.js";
 import Tasks from "./Tasks.js";
 
-const initialData = [
-    {
-        id: 517,
-        text: "Eat Lunch"
-    },
-    {
-        id: 787,
-        text: "Buy new John Grisham book"
-    },
-    {
-        id: 911,
-        text: "Buy new John Grisham book"
-    },
-    {
-        id: 231,
-        text: "Buy new John Grisham book"
-    },
-    {
-        id: 589,
-        text: "Buy new John Grisham book"
-    }
-];
+import { initializeApp } from "firebase/app";
+import { collection, deleteDoc, doc, getFirestore, query, serverTimestamp, setDoc, orderBy } from "firebase/firestore";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+
+const firebaseConfig = {
+    apiKey: "AIzaSyDxRZnwVYgJJSZOtNsJcxYGcCx-Ta_DAxk",
+    authDomain: "cs124-lab3-1ad74.firebaseapp.com",
+    projectId: "cs124-lab3-1ad74",
+    storageBucket: "cs124-lab3-1ad74.appspot.com",
+    messagingSenderId: "660026892825",
+    appId: "1:660026892825:web:780ac3094a1656d26a4088"
+};
+
+const firebaseApp = initializeApp(firebaseConfig);
+const db = getFirestore(firebaseApp);
+
+const collectionName = "cs124-lab3";
 
 function App() {
-    const [taskList, setTaskList] = useState(initialData);
-    const [completedTaskList, setCompletedTaskList] = useState([517, 911, 231]);
     const [editingTaskId, setEditingTaskId] = useState("");
     const [showAlert, setShowAlert] = useState(false);
-    const [taskToDelete, setTaskToDelete] = useState("");
+    const [showPriorityDropdown, setShowPriorityDropdown] = useState("");
+    const [showSortDropdown, setShowSortDropdown] = useState(false);
+    const [tabIndex, setTabIndex] = useState(1);
+    const [taskToDeleteParams, setTaskToDeleteParams] = useState(["", false]);
+    const [sortDirection, setSortDirection] = useState("asc");
+    const [sortParam, setSortParam] = useState("text");
+    const q = query(collection(db, collectionName), orderBy(sortParam, sortDirection));
+    const [taskList, loading] = useCollectionData(q);
+
+    function handleEditTask(taskId, field, value) {
+        setDoc(doc(db, collectionName, taskId),
+            {[field]: value},
+            {merge: true});
+        if (field === "text") {
+            handleEditingTaskIdChange(taskId);
+        }
+    }
 
-    function handleEditTask(taskId, newText) {
-        setTaskList(taskList.map(
-            task => task.id === taskId ? {...task, "text": newText} : task))
+    function handleEditingTaskIdChange(taskId) {
+        if (editingTaskId === taskId) {
+            setEditingTaskId("");
+        } else {
+            setEditingTaskId(taskId);
+        }
     }
 
     function handleSetCompletedTask(taskId) {
-        if (completedTaskList.includes(taskId)) {
-            setCompletedTaskList(completedTaskList.filter(id => id !== taskId));
+        setDoc(doc(db, collectionName, taskId),
```

Comment:
If you use updateDoc, you don't need to provide merge option.

Might have been easier to pass in the new value for completed (which is probably easily accessible by caller). That avoids this function having to look it up.